capsules/src/spi.rs

```diff
@@ -186,19 +187,17 @@ impl<'a, S: SpiMasterDevice> Driver for Spi<'a, S> {
     //   - does nothing if lock not held
     //
 
-    fn command(&self, cmd_num: usize, arg1: usize, _: AppId) -> isize {
+    fn command(&self, cmd_num: usize, arg1: usize, _: AppId) -> ReturnCode {
         match cmd_num {
-            0 /* check if present */ => 0,
+            0 /* check if present */ => ReturnCode::SUCCESS,
             // No longer supported, wrap inside a read_write_bytes
-            1 /* read_write_byte */ => -1,
+            1 /* read_write_byte */ => ReturnCode::ENOSUPPORT,
             2 /* read_write_bytes */ => {
                 if self.busy.get() {
-                    return -1;
+                    return ReturnCode::EBUSY;
                 }
-                let mut result = -1;
-                self.app.map(|app| {
+                self.app.map_or(ReturnCode::FAIL /* XXX app is null? */, |app| {
```

Comment:
I think this is the same issue as console and nrfSerialization (missing allow first)? cc @phil-levis